addon/-private/system/map-with-default.js

```diff
@@ -0,0 +1,60 @@
+import Map from './map';
+
+/*
+  This is needed because `Map` requires instantiation with `new` and by default
+  Babel transpilation will do `superConstructor.apply(this, arguments)` which
+  throws an error with native `Map`.
+
+  The desired code (if we lived in an "only native class" world) would be:
+
+  ```js
+  export default class MapWithDefault extends Map {
+    constructor(options) {
+      super();
+      this.defaultValue = options.defaultValue;
+    }
+
+    get(key) {
+      let hasValue = this.has(key);
+
+      if (hasValue) {
+        return super.get(key);
+      } else {
+        let defaultValue = this.defaultValue(key);
+        this.set(key, defaultValue);
+        return defaultValue;
+      }
+    }
+  }
+  ```
+*/
+
+export default class MapWithDefault {
```

Comment:
Should we add `copy` and `isEmpty` to warn/error with a useful message for people who may be relying on this?

It seems unlikely, but on the other hand, `MapWithDefault` is returned by some public API